controllers/discovery_stroage.go

```diff
@@ -0,0 +1,144 @@
+package controllers
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/open-policy-agent/gatekeeper/v3/apis/config/v1alpha1"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes"
+)
+
+var ErrNotFoundDiscovery = errors.New("There is no matched apiGroup, version or kind")
+
+type discoveryInfo struct {
+	apiResourceList        []*metav1.APIResourceList
+	discoveryLastRefreshed time.Time
+}
+
+type DiscoveryStorage struct {
+	discoveryInfo
```

Comment:
I don't think you need the `discoveryInfo` type anymore. I think you can just move those struct fields to `DiscoveryStorage`.